src/SigningPortal/SigningPortal.tsx

```diff
@@ -197,6 +235,38 @@ export const SigningPortal: React.FC = () => {
     setDryRunResult(result);
   };
 
+
+  const calculateCosts = async (tx: UnsafeTransaction<any, string, string, any>, api: UnsafeApi<any>) => {
+    let decodedCall = tx?.decodedCall;
+    if (!selectedAccount || (tx.decodedCall.type !== "Registrar" || (tx.decodedCall.value.type !== "reserve" && tx.decodedCall.value.type !== "register"))) {
```

Comment:
might make sense to convert you conditional to a variable or function. (this one `(tx.decodedCall.type !== "Registrar" || (tx.decodedCall.value.type !== "reserve" && tx.decodedCall.value.type !== "register")`

Used a few times and is quite long.

Reply:
Nice, refactored: https://github.com/r0gue-io/pop-wallet-signing-portal/pull/7/commits/b8254baa6f920cce505a86c6b3852cfbe51f1dda